### PR TITLE
Feat/form group page heading

### PR DESCRIPTION
**Pull request by marianayovcheva**
Added `legend_heading` to the form group options. Addresses a semantic issue of having no `h1` on 'one question per page' type of forms following gov.uk's approach described here: https://design-system.service.gov.uk/get-started/labels-legends-headings/
Set up a page for radio buttons in the `design-system-docs`.

### CODE REVIEW

File: design-system-docs/src/_component_examples/_radio_group/additional_attributes.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (marianayovcheva) [inline design-system-docs/src/_component_examples/_radio_group/additional_attributes.erb:1]: I'm not sure if/how to lint those files or if we should add a linting command later on

File: design-system-docs/src/_component_examples/_radio_group/default.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+title: Default
+---
+<%= render CitizensAdviceComponents::RadioGroup.new(
+  legend: "Example radio group",
+  name: "radio-buttons"
+) do |c|
+  c.inputs([
```
Comment (marianayovcheva) [inline design-system-docs/src/_component_examples/_radio_group/default.erb:8]: Is there a way to reuse a radio group variable the same way it's done in `radio_group_preview.rb` (i.e `c.inputs(radios)`)? I found myself copy pasting the same inputs quite a lot.

File: design-system-docs/src/_data/component_arguments.yml [hunk 3]
```diff
@@ -9,3 +9,22 @@ targeted_content:
...
   - argument: close_button
     description: Optional, include close button at the end of content. Defaults to true
+radio_group:
+  - argument: name
+    description: Required, field name
+  - argument: label
+    description: Required, the text for the label associated with the input
+  - argument: options
+    description: Optional, a hash with one or more of the following values
+  - argument: hint
```
Comment (marianayovcheva) [inline design-system-docs/src/_data/component_arguments.yml:19]: This is probably a question for a future PR but I'm not sure how to indent the arguments that are part of the options, in the old docs it is done with '&rarr; hint' but this throws an error here

File: design-system-docs/src/_component_examples/_radio_group/default.erb (hunk 2 above)
Comment (davidrapson) [reply design-system-docs/src/_component_examples/_radio_group/default.erb:8]: Possibly: You could define the options as front matter in `_component_examples/_radio_group/_defaults.yml` and use the data from each example. But I'm also fine with it being explicit for example. Makes copy-pasting the example from the docs less error prone.

File: design-system-docs/src/_component_examples/_radio_group/additional_attributes.erb (hunk 1 above)
Comment (davidrapson) [reply design-system-docs/src/_component_examples/_radio_group/additional_attributes.erb:1]: We could maybe use something like [erb-lint](https://github.com/Shopify/erb-lint) but I don't know if the front matter would confuse things. I'm inclined to not worry about linting for the docs site, especially as we want designers to be able to easily edit things.

File: design-system-docs/src/_component_docs/radio-group.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+### Additional attributes (View component only)
+
+Any additional key/value pairs (beyond `label` and `value`) in your radio button definition will be added to the input.
+
+<%= render(Shared::ComponentExample.new(:radio_group, :additional_attributes)) %>
+
+### With heading in the legend
+
+Used for one question per page forms.
```
Comment (davidrapson) [inline design-system-docs/src/_component_docs/radio-group.md:87]: Would you mind sticking a link somewhere to https://design-system.service.gov.uk/get-started/labels-legends-headings/#legends-as-page-headings to remind us why we do this?

**Review by davidrapson — CHANGES_REQUESTED** [review]
Looks great! Couple of small suggestions but fab so far. It looks like you've got a bunch of unexpected changes to the demo package lock file, mind fixing that up first?

**Comment by marianayovcheva** [comment]
@davidrapson good should about `package-lock.json` in `demo`. I've reverted it locally and tried to see at what point it started to differ. What I found was that when I was in `/demo` and running `./bin/setup` it would alter my `package-lock.json` but if I ran `npm ci` separately, it wouldn't change the lock file. Not sure why this happens though and how to prevent it from happening in the future

**Review by davidrapson — APPROVED** [review]
👍 Looks great. Mind updating the changelog too?

**Pull request merged**